webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -1,54 +1,58 @@
 import React from 'react';
 import { Modal } from 'patternfly-react';
 import PropTypes from 'prop-types';
-import ForemanModalHeader from './subcomponents/ForemanModalHeader';
-import ForemanModalFooter from './subcomponents/ForemanModalFooter';
 import ModalContext from './ForemanModalContext';
+import ForemanModalHeader from './subcomponents/ForemanModalHeader';
 import { extractModalNodes } from './helpers';
 
 const ForemanModal = props => {
-  const { isOpen, onClose, title, ...propsToPassDown } = props; // don't pass down those two as it causes PF problems
+  const { id, title, onClose, openState, children, ...propsToPassDown } = props;
   // Extract header and footer from children, if provided
   const { headerChild, footerChild, otherChildren } = extractModalNodes(
-    props.children
+    children
   );
   const context = {
-    isOpen,
+    isOpen: openState.open,
     onClose,
     title,
   };
 
+  let headerToRender = null; // if no headerChild and no title prop, then no <ForemanModalHeader>
+  if (!headerChild && title) headerToRender = <ForemanModalHeader />; // render default header with title
+  if (headerChild) headerToRender = headerChild; // render the custom header supplied as a child
+
   return (
     <ModalContext.Provider value={context}>
-      {/* Change the name of the props we are passing down to Modal to conform to Patternfly 3 api */}
       <Modal
         onHide={onClose}
-        show={isOpen}
+        show={openState.open}
         className="foreman-modal"
         {...propsToPassDown}
       >
-        {headerChild}
+        {headerToRender}
         <Modal.Body>{otherChildren}</Modal.Body>
         {footerChild}
       </Modal>
     </ModalContext.Provider>
   );
 };
 
-// Header and Footer use the provided children, or default markup if none provided
-
-ForemanModal.Header = ForemanModalHeader;
-ForemanModal.Footer = ForemanModalFooter;
-
 ForemanModal.propTypes = {
-  isOpen: PropTypes.bool.isRequired,
   children: PropTypes.node,
-  title: PropTypes.string.isRequired,
+  title: PropTypes.string,
+  id: PropTypes.string.isRequired,
+  openState: PropTypes.shape({
```

Comment:
Any reason to use `shape`? Why not `open: PropTypes.bool`?

Reply:
@sharvit It would be simpler in the component, yes.  But I won't want our Redux state to look like 
```js
myModal: true,
```
instead of 
```js
myModal: { open: true }
```

Reply:
I think it makes sense to do it and use a new selector that will return boolean. `selectIsModalOpen`

Reply:
@sharvit Ok, will update